packages/runtime/lifecycle/src/api.ts

```diff
@@ -32,33 +43,50 @@ export type ApplyInputsRequest = {
   invalidHereafter?: ISO8601;
 };
 /**
- * TODO: comment
+ * This Interface provides capabilities for runnning a Contract over Cardano.
  */
 export interface ContractsAPI {
   /**
-   * TODO: comment
+   * Submit to the Cardano Ledger, the Transaction(Tx) that will create the Marlowe Contract passed in the request.
+   * @param createContractRequest Request parameters for creating a Marlowe Contract on Cardano
    * @throws DecodingError
    */
-  create(req: CreateRequest): Promise<ContractId>;
-  // [[apply-inputs-next-provider]]
-  // DISCUSSION: Instead of providing a function from "Next" => ApplyInputsRequest, I think
-  //             this as a low level API should just receive the ApplyInputsRequest, and if we want we can
-  //             add a "higher-level" helper
-  //             Search for [apply-inputs-next-provider] for usage
+  createContract(
+    createContractRequest: CreateContractRequest
+  ): Promise<[ContractId, TxId]>;
+
   /**
-   * TODO: comment
+   * Submit to the Cardano Ledger, the Transaction(Tx) that will apply inputs to a given created contract.
+   * @param contractId Contract Id where inputs will be applied
+   * @param applyInputsRequest inputs to apply
    * @throws DecodingError
    */
   applyInputs(
     contractId: ContractId,
-    provideInput: ProvideInput
-  ): Promise<ContractId>;
+    applyInputsRequest: ApplyInputsRequest
+  ): Promise<TxId>;
+
+  /**
+   * @experimental
+   * Provide Applicability and Reducibility Information moving forward for a given contract connected to a wallet.
```

Comment:
This might be bigger than this PR, but again, I think that `reduce contract == Apply empty tx` so, Applicability implies Reducibility and `Next` was a way to ask "What can I do next?"

I think both `getApplicableInputs` and `getNextInputs` have a more apparent intent (I prefer the first one).